### ISSUE TITLE
SvelteUI FOUC in SSG mode

### ISSUE THREAD

**Issue by nxy7** [svelteuidev/svelteui#107]
### What package has an issue

@svelteuidev/core

### A clear and concise description of what the bug is

In sveltekit SSG mode initial content is sent without styling, which results in FOUC or worse - if JS is disabled components will not be rendered completely.
Discord user 'Princes' narrowed this issue to Stitches library. Probably styles from Stitches are not rendered correctly in SSG mode and therefor the components are not rendered until stitches renders styles ON THE CLIENT SIDE. 
Hint to solving this might be the fact that getCssText() is returning different values in SSR and SSG modes.

Sadly I'm not experienced enough to contribute to solving this issue.

### Steps To Reproduce

1. Create repo with SvelteUI and any component
2. Disable JS in the browser (for better visibility of the issue)
3. Change sveltekit adapter to static adapter
4. npm run build && npm run preview

### Do you know how to fix the issue

No

### Are you willing to participate in fixing this issue and create a pull request with the fix

No

**Comment by nxy7** [comment]
Turns out setting SvelteUIProvider ssr={true} fixes it completely.

**Comment by gw2princeps** [comment]
Reproduction: 

1. Open the dev server for svelteui-core
2. Add `ssr` to the svelteuiprovider 
3. Add this to `test.svelte` to enable SSG:
```
<script context="module">
	export const prerender = true;
</script>
``` 
4. Inspect the generated file in `.svelte-kit/output/prerendered`. It is missing a whole bunch of styles!

Further, run `npx svelte-kit build && npx svelte-kit preview`. There is a FOUC. After loading it once the styles are fixed. Compare the stitches style tag in the prerendered file and in the browser after loading it once, you'll notice all the stitches styles are missing.

**Comment by BeeMargarida** [comment]
This should be fixed in the latest version 0.8.0 with the hooks solution - https://www.svelteui.org/theming/ssr
